packages/dbgpt-serve/src/dbgpt_serve/auth/tests/test_runtime_access.py

```
@@ -0,0 +1,270 @@
+        )
+
+    assert service.require_document_access(user, "10").resource_id == "3"
+
+
+def test_fastapi_guard_returns_401_403_and_success(access_data):
+    app = FastAPI()
+
+    @app.get("/datasources/{resource_id}")
+    async def protected_datasource(
+        resource_id: str,
+        user: UserRequest = Depends(require_permission("DATASOURCE_USE")),
```

Ruff (0.15.21)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)